packages/core/src/hooks/call.ts

```diff
@@ -51,13 +51,13 @@ function starknetCallReducer(state: State, action: Action): State {
 }
 
 interface UseStarknetCallArgs {
-  contract?: Contract
+  contract?: ContractInterface
   method?: string
-  args?: Args
+  args?: Array<any>
```

Comment:
```suggestion
interface UseStarknetCallArgs<T> {
  contract?: ContractInterface
  method?: string
  args?: Array<T>
```

I think we can do something like this to allow the user to pass in a type

Reply:
I think the issue with this is that the arguments can be different types (bignumbers, structs, arrays, etc).

Reply:
I think the user could still chose to pass in `any` right? In which case it would be the same as it is currently, but opens the option of stricter types

Reply:
How about this? 

```suggestion
  interface UseStarknetCallArgs<T extends unknown []> {
    contract?: ContractInterface
    method?: string
    args?: T
  }
```

Then users can get stronger typing if they want:

```ts
useStarknetCall({args: [0, 'string']})
useStarknetCall<[number, string]>({args: [0, 'foo']})
```

Reply:
Oh yeah that is much nicer! I wasn't sure the exact apis when i suggested. This seems like the right way